scorep/helper.py

```diff
@@ -74,15 +78,15 @@ def add_to_ld_library_path(path):
         os.environ["LD_LIBRARY_PATH"] = ':'.join([path] + library_paths)
 
 
-def generate_compile_deps(config=[]):
+def generate_compile_deps(config):
     """
     Generates the data needed for compilation.
     """
 
-    scorep_config = ["scorep-config"] + config + ["--user"]
```

Comment:
Why did you remove the `--user`?

Reply:
I moved that up to reduce the amount of magic scorep-python is doing. This makes this helper function more flexible as the caller is responsible for adding the `--user`, see https://github.com/score-p/scorep_binding_python/pull/122/files#diff-33e0c1512b38043a56ab9c6c83e25c7c4c7c9a50283ea20b70f9e645343eafd6R117-R118

This change basically resulted from wanting to print the actual config used at https://github.com/score-p/scorep_binding_python/pull/122/files#diff-33e0c1512b38043a56ab9c6c83e25c7c4c7c9a50283ea20b70f9e645343eafd6R121

If the sub-functions add another argument, then the print is incomplete. So not touching the config afterwards ensures that the output is correct/complete and avoids the code duplication by having to add `--user` in multiple places

Reply:
Alright.